src/config/builder.rs

```diff
@@ -272,7 +272,21 @@ impl Builder {
 
     /// Build a new `RedisPool`
     pub fn finish_redis_rs(self) -> InitializationResult<RedisPool> {
-        self.finish(RedisRsFactory::new)
+        let (resolver, background_task) = trust_dns_resolver::AsyncResolver::from_system_conf()
```

Comment:
We can use [read_system_conf](https://docs.rs/trust-dns-resolver/0.18.0/trust_dns_resolver/system_conf/fn.read_system_conf.html) to read resolv.conf separately, allowing us to tweak the resolver options such as [cache_size](https://docs.rs/trust-dns-resolver/0.18.0/trust_dns_resolver/config/struct.ResolverOpts.html#structfield.cache_size)